File: cropme/src/main/java/com/takusemba/cropme/Croppable.java
package com.takusemba.cropme;

import android.graphics.Bitmap;
import android.net.Uri;
import android.widget.ImageView;

/**
 * Interface to set an Image, and return an cropped image back.
 **/
interface Croppable {

    /**
     * setUri to {@link ImageView}
     **/
    void setUri(Uri uri);

    /**
     * setBitmap to {@link ImageView}
     **/
    void setBitmap(Bitmap bitmap);

    /**
     * check if Image is out of the frame.
     */
    boolean isOffOfFrame();

    /**
     * crop image.
     **/
    void crop(OnCropListener listener);
}

File: cropme/src/main/java/com/takusemba/cropme/Croppable.kt
package com.takusemba.cropme

import android.graphics.Bitmap
import android.net.Uri
import android.widget.ImageView

import androidx.annotation.MainThread

/**
 * Interface to set an Image, and return an cropped image back.
 */
interface Croppable {

  /**
   * check if Image is out of the frame.
   */
  fun isOffFrame(): Boolean

  /**
   * setUri to [ImageView]
   */
  fun setUri(uri: Uri)

  /**
   * setBitmap to [ImageView]
   */
  fun setBitmap(bitmap: Bitmap)

  /**
   * Crop the Image on a new thread, and return the cropped Bitmap on MainThread via [OnCropListener].
   */
  @MainThread
  fun crop(listener: OnCropListener)
}
